In [53]:
import pandas as pd
import numpy as np

In [54]:
df = pd.read_csv('sales_data.csv', encoding='cp1252')
df.head(5)

Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  

[5 rows x 21 columns]

In [55]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

- No. of Rows = 9994

- No. of Columns = 21

- No null values

In [56]:
df.duplicated().sum()


np.int64(0)

- No duplicates

In [57]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')

In [58]:
df['Ship Mode'].unique()

array(['Second Class', 'Standard Class', 'First Class', 'Same Day'],
      dtype=object)

In [59]:
df['Category'].unique()

array(['Furniture', 'Office Supplies', 'Technology'], dtype=object)

In [60]:
df['Segment'].unique()


array(['Consumer', 'Corporate', 'Home Office'], dtype=object)

In [61]:
df['Region'].unique()

array(['South', 'West', 'Central', 'East'], dtype=object)

In [68]:
# turning 'order date' and 'ship date' into datetime objects
clean_df['Order Date'] = pd.to_datetime(clean_df['Order Date'])
clean_df['Ship Date'] = pd.to_datetime(clean_df['Ship Date'])

NameError: name 'clean_df' is not defined

In [63]:
print(f"Max Order Date: ", max(df['Order Date']))
print(f"Min Order Date: ", min(df['Order Date']))

Max Order Date:  2017-12-30 00:00:00
Min Order Date:  2014-01-03 00:00:00


In [64]:
print(f"Max Ship Date: ", df['Ship Date'].max())
print(f"Min Ship Date: ", df['Ship Date'].min())

Max Ship Date:  2018-01-05 00:00:00
Min Ship Date:  2014-01-07 00:00:00


In [65]:
Avg_Days_to_Ship = (df['Ship Date'] - df['Order Date']).dt.days.mean()
Max_Days_to_Ship = (df['Ship Date'] - df['Order Date']).dt.days.max()
Min_Days_to_Ship = (df['Ship Date'] - df['Order Date']).dt.days.min()
print(f"Average Days to Ship: ", Avg_Days_to_Ship)
print(f"Longest Days to Ship: ", Max_Days_to_Ship)
print(f"Shortest Days to Ship: ", Min_Days_to_Ship)

Average Days to Ship:  3.958174904942966
Longest Days to Ship:  7
Shortest Days to Ship:  0
